src/site/content/en/blog/script-component/index.md

```diff
@@ -0,0 +1,221 @@
+
+1. Use the `async` or `defer` attribute with `<script>` tags that tell the browser to load non-critical third-party scripts without blocking the document parser. Scripts not required for initial page load or the first user interaction  may be considered non-critical.
+
+    ```html
+       <script src="https://example.com/script1.js" defer></script>
+       <script src="https://example.com/script2.js" async></script> 
+    ```
+
+2. Establish early connections to required origins using preconnect and dns-prefetch. This allows critical scripts to start downloading earlier.
+
+    ```html
+       <head>
+           <link rel="preconnect" href="http://PreconnThis.com">
+           <link rel="dns-prefetch" href="http://PrefetchThis.com">
+       </head>
+    ```
+
+
+3. [Lazy-load](/embed-best-practices/#lazy-loading) third-party resources and embeds after the main page content has finished loading or when the user scrolls down to the part of the page where they are included.
+
+
+## The Next.js Script component
+
+The [Next.js Script component](https://nextjs.org/docs/basic-features/script) implements the above methods for sequencing scripts and provides a template for developers to define their loading strategy. Once the suitable strategy is specified, it will load optimally without blocking other critical resources. 
+
+The Script component builds on the HTML &lt;script> tag and provides an option to set the loading priority for third-party scripts using the strategy attribute.
+
+
+```html
+// example for beforeInteractive
+<Script src="https://polyfill.io/v3/polyfill.min.js?features=IntersectionObserverEntry%2CIntersectionObserver" strategy="beforeInteractive" />
+
+// example for afterInteractive (default)
+<Script src="https://example.com/samplescript.js" />
+
+// example for lazyonload
+<Script src="https://connect.facebook.net/en_US/sdk.js" strategy="lazyOnload" />
+```
+
+The strategy attribute can take three values.
+
+1. **beforeInteractive**: This option may be used for critical scripts that should execute before the page becomes interactive. Next.js ensures that such scripts are injected into the initial HTML on the server and executed before other self-bundled JavaScript. Consent management, bot detection scripts, or helper libraries required to render critical content are good candidates for this strategy.
+
+2. **afterInteractive**: This is the default strategy applied and is equivalent to loading a script with the defer attribute. It should be used for scripts that the browser can run after the page is interactive - for example, analytics scripts. Next.js injects these scripts on the client-side, and they run after the page is hydrated. Thus, unless otherwise specified, all third-party scripts defined using the Script component are deferred by Next.js, thereby providing a strong default.  
```

Comment:
```suggestion
2. **`afterInteractive`**: This is the default strategy applied and is equivalent to loading a script with the defer attribute. It should be used for scripts that the browser can run after the page is interactive - for example, analytics scripts. Next.js injects these scripts on the client-side, and they run after the page is hydrated. Thus, unless otherwise specified, all third-party scripts defined using the Script component are deferred by Next.js, thereby providing a strong default.  
```